sklearn/metrics/_ranking.py

```diff
@@ -381,7 +381,25 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None, max_fpr=None):
     max_area = max_fpr
     return 0.5 * (1 + (partial_auc - min_area) / (max_area - min_area))
 
-
+@validate_params(
+    {
+        "y_true": ["array-like"],
+        "y_score": ["array-like"],
+        "average": [
+            StrOptions({"micro", "macro", "samples", "weighted"}),
+            None,
+        ],
+        "sample_weight": ["array-like", None],
+        "max_fpr":[
+            Options(Real, {0.0, 0.1}),
+            None,
+        ],
+        "multi_class": [
+                        StrOptions({"raise", "ovr", "ovo"}),
+                        ],
+        "labels": ["array-like", None],
+    }
+)
```

Comment:
We will need to add this function to the list of functions to be tested in the file `sklearn/tests/test_public_functions.py`. For the moment, we are not testing the function if it is not present in the list.

https://github.com/scikit-learn/scikit-learn/blob/ec8a2a63b3e02bb955079c2af26d95af6a407242/sklearn/tests/test_public_functions.py#L112-L210